### PR TITLE
Improve zales PR

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Related tickets |  |
| License | MIT |

Small refactor moved from #6146

### CODE REVIEW

File: src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php [hunk 1]
```diff
@@ -237,4 +232,14 @@ private function getCartManager()
...
     }
+
+    /**
+     * @param OrderInterface $order
+     */
+    protected function saveCart(OrderInterface $order)
+    {
+        $cartManager = $this->getCartManager();
+        $cartManager->persist($order);
+        $cartManager->flush();
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php]: `repository->add` should be used

File: src/Sylius/Bundle/CoreBundle/EventListener/UserCartRecalculationListener.php [hunk 2]
```diff
@@ -56,10 +56,6 @@ public function recalculateCartWhileLogin(Event $event)
             return;
         }
 
-        if (!$cart instanceof OrderInterface) {
-            throw new UnexpectedTypeException($cart, OrderInterface::class);
-        }
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/EventListener/UserCartRecalculationListener.php]: Why?

Comment (lchrusciel) [reply src/Sylius/Bundle/CoreBundle/EventListener/UserCartRecalculationListener.php]: It is not a part of my PR

**Comment by Zales0123** [comment]
@lchrusciel I would be glad to merge it, after one comment fix and rebase ;)

File: src/Sylius/Bundle/CoreBundle/EventListener/UserCartRecalculationListener.php (hunk 2 above)
Comment (michalmarcinkowski) [reply src/Sylius/Bundle/CoreBundle/EventListener/UserCartRecalculationListener.php]: Ok :)

PS. It is a part of this PR because I can see it 😄

Comment (lchrusciel) [reply src/Sylius/Bundle/CoreBundle/EventListener/UserCartRecalculationListener.php]: Just a shortcut, sorry. This change is not a part of my intentional work it is a result of changes between branches. Probably @Zales0123 amend one of his commits

**Comment by lchrusciel** [comment]
@Zales0123 on it

File: src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php (hunk 1 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php]: done

**Pull request closed without merging**